README.md

```diff
@@ -327,11 +344,7 @@ async function messageHandler() {
 }
 
 // promise style:
-const pool2 = new sql.ConnectionPool(config, err => {
-    // ... error checks
-}).connect();
-
-pool2.on('error', err => {
+const pool2 = new sql.ConnectionPool(config).catch(err => {
```

Comment:
This might not be quite right, actually... Then constructor does not return a promise

Reply:
It isn't right I'm afraid, although this could be a good opportunity for scope creep so that the constructor returns a Promise if a callback function isn't specified. At the moment you get _something_ back if you define one:

https://github.com/tediousjs/node-mssql/blob/69d6a662ed5fd6183e6a2bd3e424bd53a2f7784b/lib/base.js#L129-L135

https://github.com/tediousjs/node-mssql/blob/69d6a662ed5fd6183e6a2bd3e424bd53a2f7784b/lib/base.js#L153-L155

Reply:
a constructor *surely* can't return anything other than the instance?

Reply:
🤦‍♂️ I need to do less reviews when tired